src/SFA.DAS.Commitments.Api.IntegrationTests/DatabaseSetup/Generators/CommitmentGenerator.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using AutoFixture;
+using SFA.DAS.Commitments.Api.IntegrationTests.DatabaseSetup.Entities;
+using SFA.DAS.Commitments.Api.IntegrationTests.Tests;
+
+namespace SFA.DAS.Commitments.Api.IntegrationTests.DatabaseSetup.Generators
+{
+    public class CommitmentGenerator
+    {
+        public async Task<IEnumerable<DbSetupCommitment>> Generate(long lastCohortId, long firstNewCohortId)
+        {
+            int commitmentsToGenerate = (int)(1 + lastCohortId - firstNewCohortId);
```

Comment:
this could throw and OverflowException if the summation is > int.MaxValue

Reply:
See reply to ApprenticeshipUpdateGenerator.cs comment.